### PR TITLE
Variables can have run-time-genreated random values of a configured length

### PR DESCRIPTION
**Pull request by samnissen**

### CODE REVIEW

File: app/views/data_element_values/_form.html.erb [hunk 1]
```diff
@@ -1,29 +1,89 @@
...
+  </div>
   <%= f.hidden_field :data_element_id, :value => @data_element_value.data_element_id %>
 
+  <div class="form-group variable-control-container">
+    <%= f.label :random_value, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
+      <%= f.check_box :random_value,
+          :class => 'text-field form-control',
+          :id => 'variable-control',
+          :checked => @data_element_value.random_value %>
```
Comment (jhulme) [inline app/views/data_element_values/_form.html.erb:36]: Checkbox appears to be quite large and should probably be left adjusted to align with the other elements on the page.

File: app/views/data_element_values/_form.html.erb [hunk 2]
```diff
@@ -1,29 +1,89 @@
...
+    $(".variable-control-container").click(function(i, obj){
+      showHideVariableControls();
+    });
+
+    removeUnusedVariable = function(e) {
+      e.preventDefault;
+      e.stopPropagation;
+
+      console.log('hello, friend!');
+
```
Comment (jhulme) [inline app/views/data_element_values/_form.html.erb]: Trim this line or edit the console.log output

File: app/views/data_element_values/_form.html.erb [hunk 3]
```diff
@@ -1,29 +1,89 @@
...
+    <%= f.label :value, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
       <%= f.text_field :value, :class => 'text-field form-control' %>
     </div>
   </div>
+  <div class="form-group variable-control-container" id="variable-length-entry">
+    <%= f.label :random_value_length, :class => 'control-label col-md-3' %>
+    <div class="col-md-9">
+      <%= f.number_field :random_value_length,
+          :class => 'text-field form-control' %>
```
Comment (jhulme) [inline app/views/data_element_values/_form.html.erb]: It's possible to exceed the max length of 256. Submitting a value this large returns an error screen but the alignment is out.

Preference would be to cap input here at the min/max values allowed.

File: app/views/data_element_values/_form.html.erb (hunk 1 above)
Comment (samnissen) [inline app/views/data_element_values/_form.html.erb:36]: @jhulme Here's what this looks like to me: 
screen shot 2016-04-29 at 8 28 44 am

**Comment by samnissen** [comment]
I moved the check box over in Firefox, but didn't change the size as it is reasonable in Chrome, Safari and Firefox Mac (so just not Firefox on Windows so far).

**Comment by samnissen** [comment]
Changes exist in #73

**Pull request closed without merging**